design/bridge_design.md

```diff
@@ -0,0 +1,317 @@
+![Control Diagram Not Available](./resources/RollupDiagram.svg)
+
+**If the canonical block remains the one without the message, then the enclave should regenerate a rollup from there by reapplying the transactions!**
+
+> **_NOTE:_** The following section is only for fast finality that does not support block reorgs
+
+
+<details>
+  <summary> More on hard & fast finality </summary>
+
+**L2 to L1 Delivery** - The `ManagementContract` needs to ensure that information is exposed through the APIs and/or messages are submitted only after the challenge period of a rollup has passed. This means that anything that shows up there will be final.
+
+**L1 to L2 Delivery** - We need to implement a block confirmation mechanism to secure the `MessageBus` - when the enclave encounters the `LogMessagePublished` it will only create a synthetic transaction for it on the L2 once the required `consistencyLevel` is achieved. Otherwise if we publish the message instantly it might end being fake as the `L1` event that caused it has been reogranized.
+
+We can also engineer a mechanism to insure delivered messages:
+* When fees are collected for `publishMessage`, part of those fees is funneled to an insurance pool.
+* When a message that has been published gets reorganized the ObscuroDAO can pay for the financial loss from the insurance pool.
+
+> **_NOTE:_** The maximum possible payout should always be less than the penalty for PoS slashing on MainNet. This ensures there is no profit to attacking and claiming insurance.  
+
+</details>
+
+
+
+## Bridge
+
+The actual bridge will be deployed as a [`Proxy`](https://docs.openzeppelin.com/contracts/4.x/api/proxy).
+
+The initial version of the bridge implementation is going to create wrapped versions of ERC tokens. The bridge will hook into the `MessageBus` API in order to provide its functionality.
+
+It will have the following functions:
+* ```solidity 
+  function bridgeAssets(address erc20token, uin256 amount) public
+  ```
+* ```solidity
+  function receiveAssets(bytes calldata memory) public
+  ```
+* ```solidity
+  function bridgeETH() payable public //this function will only be available on L1 as L2 will only have WETH
+  ``` 
+* functions to interact with OpenZeppelin's whitelist.
+
+It will also have the following properties required for the OpenZeppelin whitelist to run. The whitelist is discussed in detail further down this document.
+
+
+
+
+### Cross-chain messages definitions
+
+There will be the following topics initially: 
+* `Transfers` - messages here will be for users transferring value across chains.
+* `Administrative` - messages here will be for administrative changes; Voted upgrades, and whitelist changes.
+
+
+### Transfers
+
+**Diagram of the bridge process that happens on the Ethereum MainNet**
+![Sequence Diagram not available](./resources/MainNet_MessageBus_Diagram.svg)
+
+---
+
+**Diagram of the bridge process that happens on Obscuro's Layer 2**
```

Comment:
this diagram should be split up. 
It contains both the deposits and the withdrawal sequencer, and is a bit confuxing